app/validation/validators.py

```diff
@@ -66,7 +66,7 @@ def __init__(self, min=None, max=None, messages=None):
         self.messages = messages
 
     def __call__(self, form, field):
-        data = field.data
+        data = int(field.data)
```

Comment:
I think that is because the current TimeInputCheck isn't using the validation chain and is instead calling validators manually, which means the validation chain isn't stopped after failing a cast to integer as part of the IntegerCheck. If the validators are moved to the form, this cast shouldn't be necessary.